src/CompileRun.ts

```diff
@@ -17,19 +17,17 @@ export class CompileRun {
         this.terminal = VSCodeUI.compileRunTerminal;
     }
 
-    private async compile(currentFile: string, outputFile: string, doRun: boolean = false, withFlags: boolean = false) {
+    private async compile(currentFile: vscode.TextDocument, outputFile: string, doRun: boolean = false, withFlags: boolean = false) {
         const spawn = require('child_process').spawn;
         let commandExistsSync = require('command-exists').sync;
 
-        let save = Settings.saveBeforeCompile;
-
-        if (save) {
```

Comment:
save here is a function
```typescript
let save : () => boolean = Settings.saveBeforeCompile;
```

and

```typescript
if (save) {
```
is always true.
You may mean save is the return value of the function `saveBeforeCompile`.